tchannel-core/src/main/java/com/uber/tchannel/api/handlers/JSONRequestHandler.java

```diff
@@ -20,22 +20,51 @@
  * THE SOFTWARE.
  */
 
-package com.uber.tchannel.api;
+package com.uber.tchannel.api.handlers;
 
 import com.google.common.reflect.TypeToken;
+import com.uber.tchannel.api.ResponseCode;
+import com.uber.tchannel.schemes.JSONSerializer;
+import com.uber.tchannel.schemes.RawRequest;
+import com.uber.tchannel.schemes.RawResponse;
 
-public abstract class DefaultRequestHandler<T, U> implements RequestHandler<T, U> {
+import java.util.Map;
+
+public abstract class JSONRequestHandler<T, U> implements RequestHandler {
 
     private TypeToken<T> requestType = new TypeToken<T>(getClass()) { };
```

Comment:
Can we move back to having both a RequestHandler interface and a DefaultRequestHandler for handling some of the shared logic e.g. getting the generic response types?